crates/sui-core/src/authority_server.rs

```diff
@@ -388,11 +388,6 @@ impl ValidatorService {
         );
 
         // CRITICAL! Validators should never sign an external system transaction.
-        fp_ensure!(
-            !certificate.is_system_tx(),
-            SuiError::InvalidSystemTransaction.into()
-        );
-
```

Comment:
i agree this is probably covered elsewhere, but its such a cheap check i would lean towards leaving it in just in case this code gets refactored. (i verified we do have tests though).

i think a good time to simplify this would be if/when we switch to a newtype pattern for validation.

Reply:
Sounds good, will close the PR for now